File: DataStructure/数据结构及算法实现/reviewAlgorithm/src/algorithm_imp/sorting_imp/BubbleSort.java
package algorithm_imp.sorting_imp;

/**
 * @author MrkWinter
 * @version 1.0
 */

import java.util.Arrays;

/**
 * 1. 冒泡排序
 */
public class BubbleSort {
    public static void main(String[] args) {
        int[] arr = {2,4,1,5,8,3,7,9};
        bubbleSorting(arr);
        System.out.println(Arrays.toString(arr));
    }

    public static void bubbleSorting(int[] arr) {
        for (int i = 0; i < arr.length - 1; i++) {
            boolean flag = true; //可以退出
            for (int j = 0; j < arr.length - 1 - i; j++) {
                if (arr[j] > arr[j + 1]) {
                    flag = false;
                    arr[j] = arr[j] ^ arr[j + 1];
                    arr[j+1] = arr[j] ^ arr[j + 1];
                    arr[j] = arr[j] ^ arr[j + 1];
                }
            }
            if (flag)
                return;
        }
    }
}

File: HomeWork/Python/homework02/com_3/BubbleSort.py
def bubble_sort(arr: list) -> list:
    length = len(arr)
    i = 0
    while i < length - 1:
        flag = True
        j = 0
        length2 = length - i - 1
        while j < length2:
            if arr[j] > arr[j + 1]:
                arr[j], arr[j + 1] = arr[j + 1], arr[j]
                flag = False
            j += 1
        if flag:
            return arr
        i += 1

    return arr


if __name__ == '__main__':
    my_list = bubble_sort([1, 6, 3, 4, 5, 0])
    print(my_list)
